applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_interface_element.cpp

```diff
@@ -1869,20 +1867,22 @@ void UPwSmallStrainInterfaceElement<TDim, TNumNodes>::CalculateAndAddCouplingMat
 
     const Matrix b_matrix = prod(rVariables.RotationMatrix, rVariables.Nu);
 
-    noalias(rVariables.UPMatrix) = GeoTransportEquationUtilities::CalculateCouplingMatrix(
-        b_matrix, rVariables.VoigtVector, rVariables.Np, rVariables.BiotCoefficient,
-        rVariables.BishopCoefficient, rVariables.IntegrationCoefficient);
+    BoundedMatrix<double, TNumNodes * TDim, TNumNodes> coupling_matrix =
+        GeoTransportEquationUtilities::CalculateCouplingMatrix(
+            b_matrix, rVariables.VoigtVector, rVariables.Np, rVariables.BiotCoefficient,
+            rVariables.BishopCoefficient, rVariables.IntegrationCoefficient);
 
     // Distribute coupling block matrix into the elemental matrix
-    GeoElementUtilities::AssembleUPBlockMatrix(rLeftHandSideMatrix, rVariables.UPMatrix);
+    GeoElementUtilities::AssembleUPBlockMatrix(rLeftHandSideMatrix, coupling_matrix);
 
     if (!rVariables.IgnoreUndrained) {
         const double SaturationCoefficient = rVariables.DegreeOfSaturation / rVariables.BishopCoefficient;
-        noalias(rVariables.PUMatrix) = PORE_PRESSURE_SIGN_FACTOR * SaturationCoefficient *
-                                       rVariables.VelocityCoefficient * trans(rVariables.UPMatrix);
+        BoundedMatrix<double, TNumNodes, TNumNodes * TDim> undrained_coupling_matrix =
```

Comment:
The name transposed_coupling_matrix that you chose somewhere else is more appropriate.
A coupling matrix never is undrained and in future we will try to find a better name for the wrong description IgnoreUndrained.

Reply:
I was happy to invent that name. ;) No changed names to `undrained_terms`